vsintegration/src/FSharp.Editor/CodeLens/AbstractCodeLensDisplayService.fs

```diff
@@ -0,0 +1,273 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace rec Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Windows.Controls
+open Microsoft.VisualStudio.Text
+open Microsoft.VisualStudio.Text.Editor
+open Microsoft.VisualStudio.Text.Formatting
+open System.Threading
+open System.Windows
+open System.Collections.Generic
+
+open Microsoft.VisualStudio.FSharp.Editor.Logging
+
+[<AbstractClass>]
+type CodeLensDisplayService (view : IWpfTextView, buffer : ITextBuffer, layerName) as self =
+    
+    // Add buffer changed event handler
+    do (
+        buffer.Changed.Add self.handleBufferChanged
+        view.LayoutChanged.Add self.handleLayoutChanged
+       )
+    
+    /// <summary>
+    /// Enqueing an unit signals to the tagger that all visible line lens must be layouted again,
+    /// to respect single line changes.
+    /// </summary>
+    member val RelayoutRequested : Queue<_> = Queue() with get
```

Comment:
You should typically use "let" and later publish the values using a "member" property.  It is rare to use "member val" in such a way in a large class